brew: no such file or directory

When I follow the readme to install this repo on a fresh install of MacOS Sonoma (Macbook M1), I get the following error:

```
% make
sudo -v
while true; do sudo -n true; sleep 60; kill -0 "$" || exit; done 2>/dev/null &
is-executable brew || curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install.sh | bash
if ! grep -q bash /private/etc/shells; then \
		brew install bash bash-completion@2 pcre && \
		sudo append bash /private/etc/shells && \
		chsh -s bash; \
	fi
brew install git git-extras
make: brew: No such file or directory
make: *** [git] Error 1
```

I'm having a heck of a time troubleshooting this because it makes no sense. If I change `brew` to `/opt/homebrew/bin/brew`, it works. If I print the `PATH` immediately before the Makefile calls brew, it shows `/opt/homebrew/bin` in the path. If I add `which brew` to the Makefile just before `brew`, _it correctly shows `/opt/homebrew/bin/brew`_. And yet... when the Makefile actually tries to run brew, you get `No such file or directory` 🤷

--- a/Makefile
+++ b/Makefile
@@ -1,9 +1,9 @@
-SHELL = /bin/bash
 DOTFILES_DIR := $(shell dirname $(realpath $(firstword $(MAKEFILE_LIST))))
 OS := $(shell bin/is-supported bin/is-macos macos linux)
 HOMEBREW_PREFIX := $(shell bin/is-supported bin/is-macos $(shell bin/is-supported bin/is-arm64 /opt/homebrew /usr/local) /home/linuxbrew/.linuxbrew)
 export N_PREFIX = $(HOME)/.n
 PATH := $(HOMEBREW_PREFIX)/bin:$(DOTFILES_DIR)/bin:$(N_PREFIX)/bin:$(PATH)
+SHELL := env PATH=$(PATH) /bin/bash
 SHELLS := /private/etc/shells
 BIN := $(HOMEBREW_PREFIX)/bin
 export XDG_CONFIG_HOME = $(HOME)/.config

